src/scenic/core/scenarios.py

```diff
@@ -309,8 +323,48 @@ def generate(self, maxIterations=2000, verbosity=0, feedback=None):
         Raises:
             `RejectionException`: if no valid sample is found in **maxIterations** iterations.
         """
+        scenes, iterations = self.generateBatch(1, maxIterations, verbosity, feedback)
+        return scenes[0], iterations
+
+    def generateBatch(self, numScenes, maxIterations=float('inf'), verbosity=0, feedback=None):
+        """Sample several `Scene` objects from this scenario.
+
+        For a description of how scene generation is done, see `scene generation`.
+
+        Args:
+            numScenes (int): Number of scenes to generate.
+            maxIterations (int): Maximum number of rejection sampling iterations (over all scenes).
+            verbosity (int): Verbosity level.
+            feedback (float): Feedback to pass to external samplers doing active sampling.
+                See :mod:`scenic.core.external_params`.
+
+        Returns:
+            A pair with a list of the sampled `Scene` objects and the total number
+            of iterations used.
+
+        Raises:
+            `RejectionException`: if not enough valid samples are found in **maxIterations** iterations.
+        """
+        total_iterations = 0
+        scenes = []
+
+        for _ in range(numScenes):
+            try:
+                scene, iterations = self._generateInner(maxIterations, verbosity, feedback)
+                scenes.append(scene)
+                total_iterations += iterations
```

Comment:
Need to also decrease `maxIterations` by `iterations` if `maxIterations` is supposed to be a total over all the scenes (as stated in the docstring).